isSymbol

Checks if value is classified as a Symbol primitive or object.

**Possible implementation**:
```js
const isSymbol =  curry1((val) => {
    return typeof val === 'symbol'
        || typeof val === 'object' && type(val) === 'Symbol';
});
```
[type](https://ramdajs.com/docs/#type) function is from ramda.

[This PR](https://github.com/char0n/ramda-adjunct/pull/1067) should serve as a model template of PR for `isSymbol` function.

Hi @char0n . I was hoping to tackle this issue, but I am getting an error while trying to `npm install`. Seems like the version of `sinon` in the `package-lock.json` is not available anymore 🤔 . Any thoughts on that?
```bash
npm ERR! code E404
npm ERR! 404 Not Found: sinon@7.4.0
```

--- a/src/index.js
+++ b/src/index.js
@@ -75,6 +75,7 @@ export { default as isFalsy } from './isFalsy';
 export { default as isRegExp } from './isRegExp';
 export { default as isNotRegExp } from './isNotRegExp';
 export { default as isSparseArray } from './isSparseArray';
+export { default as isSymbol } from './isSymbol';
 // Function
 export { default as stubUndefined } from './stubUndefined';
 export { default as stubNull } from './stubNull';

--- a/src/isSymbol.js
+++ b/src/isSymbol.js
@@ -1,0 +1,30 @@
+import { type } from 'ramda';
+import curry1 from 'ramda/src/internal/_curry1';
+
+/**
+ * Checks if input value is a Symbol.
+ *
+ * @func isSymbol
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.22.0|v2.22.0}
+ * @category Type
+ * @sig * -> Boolean
+ * @param {*} val The value to test
+ * @return {boolean}
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol|MDN Symbol}
+ * @example
+ *
+ * RA.isSymbol(Symbol('1')); //=> true
+ * RA.isSymbol(Symbol(1)); //=> true
+ * RA.isSymbol('string'); //=> false
+ * RA.isSymbol(undefined); //=> false
+ * RA.isSymbol(null); //=> false
+ */
+const isSymbol = curry1(val => {
+  return (
+    typeof val === 'symbol' ||
+    (typeof val === 'object' && type(val) === 'Symbol')
+  );
+});
+
+export default isSymbol;

--- a/test/isSymbol.js
+++ b/test/isSymbol.js
@@ -1,0 +1,34 @@
+import { assert } from 'chai';
+import * as R from 'ramda';
+
+import * as RA from '../src';
+
+describe('isSymbol', function() {
+  context('given a Symbol as value', function() {
+    it('should return true', function() {
+      assert.isTrue(RA.isSymbol(Symbol('any')));
+      assert.isTrue(RA.isSymbol(Symbol(1)));
+      assert.isTrue(RA.isSymbol(Symbol.iterator));
+    });
+  });
+
+  context('given anything else as value', function() {
+    it('should return false', function() {
+      assert.isFalse(RA.isSymbol(undefined));
+      assert.isFalse(RA.isSymbol(null));
+      assert.isFalse(RA.isSymbol(() => {}));
+      assert.isFalse(RA.isSymbol([]));
+      assert.isFalse(RA.isSymbol({}));
+      assert.isFalse(RA.isSymbol(NaN));
+      assert.isFalse(RA.isSymbol(true));
+      assert.isFalse(RA.isSymbol('string'));
+      assert.isFalse(RA.isSymbol(1));
+    });
+  });
+
+  it('should support placeholder to specify "gaps"', function() {
+    const isSymbol = RA.isSymbol(R.__);
+
+    assert.isFalse(isSymbol('abc'));
+  });
+});

--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -116,6 +116,11 @@ declare namespace RamdaAdjunct {
         isNotUndefined(val: any): boolean;
 
         /**
+         * Checks if input value is `Symbol`.
+         */
+        isSymbol(val: any): val is Symbol;
+
+        /**
          * Checks if input value is `null`.
          */
         isNull(val: any): val is null;

--- a/types/test/isSymbol.ts
+++ b/types/test/isSymbol.ts
@@ -1,0 +1,13 @@
+import * as RA from 'ramda-adjunct';
+
+RA.isSymbol(undefined); // $ExpectType boolean
+RA.isSymbol(null); // $ExpectType boolean
+RA.isSymbol(() => {}); // $ExpectType boolean
+RA.isSymbol(NaN); // $ExpectType boolean
+RA.isSymbol(true); // $ExpectType boolean
+RA.isSymbol('string'); // $ExpectType boolean
+RA.isSymbol(1); // $ExpectType boolean
+RA.isSymbol(Symbol('any')); // $ExpectType boolean
+RA.isSymbol(Symbol(1)); // $ExpectType boolean
+RA.isSymbol(Symbol()); // $ExpectType boolean
+RA.isSymbol(Symbol.iterator); // $ExpectType boolean

